asteroid_data_processing.py

```diff
@@ -0,0 +1,417 @@
+def separate_data(api_response, name, location):
+    match = re.search(r"\$\$SOE.*?\$\$EOE", api_response.text, re.DOTALL)
+    if match:
+        fragment = match.group()
+        lines = fragment.strip().split("\n")
+        asteroid_positions = []
+
+        for line in lines:
+            line = line.strip()
+            parts = line.split()
+            if len(parts) >= 2:
+                date_time = parts[0] + " " + parts[1]
+                datetime_obj = datetime.strptime(date_time, "%Y-%b-%d %H:%M")
+
+                # Tworzenie obiektu Time z obiektu datetime
+                time = Time(datetime_obj)
+                ra = parts[2] + " " + parts[3] + " " + parts[4]
+                dec = parts[5] + " " + parts[6] + " " + parts[7]
+                ra = better_pos_ra(ra)
+                dec = better_pos_dec(dec)
+                alt = get_altitude(ra, dec, time, location)
+                asteroid_positions.append([name, time, ra, dec, alt])
+        return np.array(asteroid_positions)
+    else:
+        logging.exception("No ephemeris for target")
+        return []
+
+
+def better_pos_ra(pos):
+    parts = pos.split()
+    hours = parts[0]
+    minutes = parts[1]
+    seconds = parts[2]
+
+    formatted_pos = f"{hours}h{minutes}m{seconds}s"
+    return formatted_pos
+
+
+def better_pos_dec(pos):
+    parts = pos.split()
+    hours = parts[0]
+    minutes = parts[1]
+    seconds = parts[2]
+
+    formatted_pos = f"{hours}d{minutes}m{seconds}s"
+    return formatted_pos
+
+
+def get_radius(x, y, x_mean, y_mean, RADIUS_FACTOR):
+    radius = (np.sqrt((x[0] - x_mean) ** 2 + (y[0] - y_mean) ** 2)) * RADIUS_FACTOR
+    logging.info(f"radius = {radius}")
+    return radius
+
+
+def get_cartesian_positions(asteroid_positions):
+    arr = asteroid_positions
+    asteroid_cartesian_positions = SkyCoord(
+        arr[:, 2], arr[:, 3], unit=(u.hourangle, u.deg)
+    )
+
+    x = asteroid_cartesian_positions.ra.deg
+    y = asteroid_cartesian_positions.dec.deg
+
+    x_mean = np.mean(x)
+    y_mean = np.mean(y)
+    return x, y, x_mean, y_mean
+
+
+@timeit
+def get_altitude(ra, dec, observing_time, observing_location):
+    coord = SkyCoord(ra, dec)
+    altaz = coord.transform_to(
+        AltAz(obstime=observing_time, location=observing_location)
+    )
+    altitude_rad = altaz.alt
+    altitude_deg = altitude_rad.to(u.deg)
+
+    return altitude_deg
+
+
+@timeit
+def get_position(start_time, stop_time, name, location):
+    session = requests.Session()
+    url = "https://ssd.jpl.nasa.gov/api/horizons.api"
+    response = session.get(
+        url,
+        params={
+            "format": "text",
+            "COMMAND": name,
+            "OBJ_DATA": "NO",
+            "EPHEM_TYPE": "OBSERVER",
+            "START_TIME": start_time,
+            "STOP_TIME": stop_time,
+            "STEP_SIZE": "1m",
+            "QUANTITIES": "1",
+        },
+    )
+    if response.status_code != 200:
+        logging.exception("Horizons API error")
```

Comment:
Here you log an exception but the code continues! You should either raise or return as processing the response body is wrong.
```suggestion
    response.raise_for_status()
```